src/Morphir/TypeScript/PrettyPrinter.elm

```diff
@@ -21,18 +26,168 @@ type alias Options =
     }
 
 
+nameSpaceNameFromPackageAndModule : Path -> Path -> String
+nameSpaceNameFromPackageAndModule packagePath modulePath =
+    (packagePath ++ modulePath)
+        |> List.map Name.toTitleCase
+        |> String.join "_"
+
+
 {-| -}
 mapCompilationUnit : Options -> CompilationUnit -> Doc
 mapCompilationUnit opt cu =
-    concat
-        [ "// Generated by morphir-elm"
-        , newLine
-        , newLine
-        , cu.typeDefs
-            |> List.map (mapTypeDef opt)
-            |> String.join (newLine ++ newLine)
-        , newLine
-        ]
+    case cu of
+        { dirPath, fileName, packagePath, modulePath, typeDefs } ->
+            concat
+                [ "// Generated by morphir-elm"
+                , newLine ++ newLine
+                , typeDefs
+                    |> List.concatMap collectRefsFromTypeDef
+                    |> renderImports packagePath modulePath
+                    |> String.join newLine
+                , newLine ++ newLine
+                , typeDefs
+                    |> List.map (mapTypeDef opt)
+                    |> List.map (\mappedTypeDef -> mappedTypeDef ++ newLine ++ newLine)
+                    |> String.join newLine
+                ]
+
+
+collectRefsFromTypeDef : TypeDef -> List NamespacePath
```

Comment:
Maybe this and other functions related with generating the top-level imports file could be moved into a submodule, to keep the PrettyPrinter simpler as a whole

Reply:
I think these actually belong in the backend, as discussed in one of the comments above. Either way, I agree on pulling them into a different file than the main PrettyPrinter module.

Reply:
Sounds good.